### ISSUE TITLE
Unable to enroll beat when using non-standard Kibana port

### ISSUE THREAD

**Issue by Rukas** [elastic/beats#9385]
We have set up Kibana to listen on port 443 over HTTPs and when trying to enroll a beat the command fails and appears to be trying to connect on port 5601.  I've only tested this with Metricbeat so far

- Version: 6.5.1 (Elasticsearch, Kibana, and beats)

```
metricbeat.exe enroll https://testurl.com 0101c2c86c2e4abeae2071e6df5ee365
Error while enrolling: fail to execute the HTTP POST request: Post https://testurl.com:5601/api/beats/agent/9bfa443f-d5bc-45c7-9a1d-8f60326d937a: dial tcp 192.168.1.134:5601: connectex: No connection could be made because the target machine actively refused it.
```

Attempting to set the port results in the following:
```
metricbeat.exe enroll https://testurl.com:443 0101c2c86c2e4abeae2071e6df5ee365
Error while enrolling: Invalid enrollment token
```

**Comment by ph** [comment]
@Rukas I've looked at the code and we add by default the port to the URL when none are specified. When specifying the port `metricbeat.exe enroll https://testurl.com:443 0101c2c86c2e4abeae2071e6df5ee365` it should connect to CM.

The message **Invalid enrollment token** is coming directly from Kibana so this mean it tried to validate the token. Can you retry with a freshly created token? Tokens expire after 10 mins.

**Comment by Rukas** [comment]
@ph I just tried to run `metricbeat.exe enroll https://testurl.com:443 <token>` again and I still get 
`Error while enrolling: Invalid enrollment token`

This was run within a minute of starting the enrollment process in Kibana

Kibana shows the following in the logs after running the enroll command:

**Comment by ph** [comment]
@Rukas Thanks for the update, I think I might know what is going on:

Looking at the URL "/api/beats/agent/unknown/53390e16acfb4e038007455e426876f7", I see an `unknown` value and this value should be the **UUID** of the agent. 

Do you have a `meta.json` file in the **data** directory?

**Comment by Rukas** [comment]
@ph 
Yes, the contents are: 

```
{"uuid":"9bfa443f-d5bc-45c7-9a1d-8f60326d937a"}
```

**Comment by ph** [comment]
Thanks @Rukas, I will take a deeper look

**Comment by ph** [comment]
@Rukas I am having a hard time to figure out where the `unknown` value in the path `/api/beats/agent/unknown/53390e16acfb4e038007455e426876f7` is coming from, did you replace that information in the comment?

The uuid generation is synchronous and it should be generated way before the enroll command is executed. Also if the value was a uuid.NIL (default) it would have returned a string exactly like this `00000000-0000-0000-0000-000000000000`.

Could do the following:
1. delete the `data/meta.json`
2. create a new enrollment token
3. prefix the enroll subcommand with the flags `-e -v -d "*"` to get full debug output.

I've searched throught the code of the uuid lib and they are no `unknown` word in the code.

**Comment by ph** [comment]
Also the more I look at it I think its **not the right log statement** with the enroll subcommand because the user agent is a Chrom which should not be the default agent from the go http library.

```
"user-agent": "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/70.0.3538.110 Safari/537.36",
```

The other that doesn't match is it should be a "POST" not a "GET" method on that endpoint.

```
		"method": "get",
```

**Comment by thomas-mullaly** [comment]
Hi @ph,

I work with @Rukas. He's busy with other things today, but I was able to retry the enrollment following your instructions.

Filebeat output (after I deleted the existing meta.json file):
```
$ filebeat enroll -e -v -d "*"  https://testurl.com:443 71c49af2913a4d39abfe3f83977a987c
2018-12-06T15:28:58.605Z        INFO    instance/beat.go:592    Home path: [/usr/share/filebeat] Config path: [/etc/filebeat] Data path: [/var/lib/filebeat] Logs path: [/var/log/filebeat]
2018-12-06T15:28:58.605Z        DEBUG   [beat]  instance/beat.go:629    Beat metadata path: /var/lib/filebeat/meta.json
2018-12-06T15:28:58.637Z        INFO    instance/beat.go:599    Beat UUID: 52af2ebf-19a7-4d6c-a932-445a3f71025e
2018-12-06T15:28:58.637Z        INFO    kibana/client.go:118    Kibana url: https://testurl.com:443
Error while enrolling: Invalid enrollment token
```

I was able to find the correct kibana log entry from when the beat tried to enroll:

```
{
    "type": "response",
    "@timestamp": "2018-12-06T15:28:58Z",
    "tags": [],
    "pid": 27186,
    "method": "post",
    "statusCode": 400,
    "req": {
        "url": "/api/beats/agent/52af2ebf-19a7-4d6c-a932-445a3f71025e",
        "method": "post",
        "headers": {
            "host": "testurl.com:443",
            "user-agent": "Go-http-client/1.1",
            "content-length": "88",
            "accept": "application/json",
            "content-type": "application/json",
            "kbn-beats-enrollment-token": "71c49af2913a4d39abfe3f83977a987c",
            "kbn-version": "",
            "accept-encoding": "gzip"
        },
        "remoteAddress": "10.0.0.9",
        "userAgent": "10.0.0.9"
    },
    "res": {
        "statusCode": 400,
        "responseTime": 151,
        "contentLength": 9
    },
    "message": "POST /api/beats/agent/52af2ebf-19a7-4d6c-a932-445a3f71025e 400 151ms - 9.0B"
}
```

Let me know if you need anything else.

**Comment by thomas-mullaly** [comment]
I dug into the Kibana server side logic for verifying if the enrollment token is valid, and I noticed tokens are stored in an index called `.management-beats`. I've looked at all of the indexes in our cluster, and that one doesn't exist. What's more weird is that our cluster _does have_ the `.management-beats` index template.

I tried to manually create the index, but got the following error message:
```Rejecting mapping update to [.management-beats] as the final mapping would have more than 1 type: [_doc, doc]```

**Comment by ph** [comment]
@asjadathick indeed this is the problem in this case, will check on our side to improve the behavior.

**Comment by mattapperson** [comment]
Yes, you need to keep adding indexes until you have what’s needed. `.management-beats` is all that we need, but other solutions might need their indexes added too, not really sure why those are throwing issues only when you add our index pattern.

As for the url with unknown in it, that’s kibana making a call to the api to ping for the beat using the enrollment token to find it as the beats I’d is unknown

**Comment by thomas-mullaly** [comment]
@mattapperson I ended having to set the config value to:
```
action.auto_create_index: ".management-beats,.security_audit_log*,.watches,.triggered_watches,.watcher-history-*"
```
in order to get the node to start properly.

I rolled this change out to all of the other nodes in our cluster and restarted our kibana instance for good measure. Unfortunately, the `.management-beats` index is still not being created and trying to enroll a beat ends with the same "Error while enrolling: Invalid enrollment token" message. I double checked the docs to see if I had missed any elasticsearch/kibana steps for getting central management setup, but it seems like the enrollment should work out of the box.

Not sure what I should try next to get this to work.

**Comment by mattapperson** [comment]
@thomas-mullaly are you able to post your config here (redacted of sensitive content) so I can better try and replicate your issue?

**Comment by mattapperson** [comment]
@thomas-mullaly Also ensuring you restarted Kibana after this change to your ES config?

**Comment by thomas-mullaly** [comment]
@mattapperson 

Here's our config settings:

**elasticsearch.yml**
```
cluster.name: prd-cluster

node.name: ${HOSTNAME}

node.attr.box_type: "warm"

path:
  data: 
    - /data/elasticsearch/data
    - /data1/elasticsearch/data
    - /data2/elasticsearch/data
  logs: /data/elasticsearch/logs

network.host: 0.0.0.0

discovery.zen.ping.unicast.hosts: ["10.0.0.7","10.0.0.9","10.0.0.10"]

discovery.zen.minimum_master_nodes: 2

script.painless.regex.enabled: true
action.auto_create_index: ".management-beats,.security_audit_log*,.watches,.triggered_watches,.watcher-history-*"

cloud.azure.storage.my_account.account: <redacted>
cloud.azure.storage.my_account.key: <redacted>

xpack.security.enabled: true
xpack.monitoring.enabled: true
xpack.graph.enabled: true
xpack.watcher.enabled: true
xpack.ml.enabled: true

xpack.security.transport.ssl.enabled: true
xpack.security.transport.ssl.verification_mode: certificate
xpack.security.transport.ssl.keystore.path: certs/elastic-certificates.p12
xpack.security.transport.ssl.truststore.path: certs/elastic-certificates.p12

xpack.security.http.ssl.enabled: true
xpack.security.http.ssl.keystore.path: certs/elastic-certificates.p12
xpack.security.http.ssl.truststore.path: certs/elastic-certificates.p12

xpack.security.authc.token.enabled: true
xpack.security.authc.realms:
  azure:
    type: saml
    order: 1
    idp.metadata.path: elasticsearch-saml.xml
    idp.entity_id: "<redacted>"
    idp.use_single_logout: false
    sp.entity_id:  "https://<redacted>/"
    sp.acs: "https://<redacted>:443/api/security/v1/saml"
    sp.logout: "https://<redacted>/logout"
    attributes.principal: "http://schemas.xmlsoap.org/ws/2005/05/identity/claims/emailaddress"
    attributes.groups: "http://schemas.xmlsoap.org/ws/2005/05/identity/claims/assignedroles"
    attributes.name: "http://schemas.xmlsoap.org/ws/2005/05/identity/claims/name"
    attributes.mail: "http://schemas.xmlsoap.org/ws/2005/05/identity/claims/emailaddress"
  native1:
    type: native
    order: 2

xpack.security.audit.enabled: true
xpack.security.audit.outputs: [index]
xpack.security.audit.index.events.exclude: ["access_granted"]
xpack.security.audit.index.settings:
  index:
    number_of_shards: 3
    number_of_replicas: 1
xpack.security.audit.index.rollover: weekly
```

**kibana.yml**
```
server.port: 443
server.host: "0.0.0.0"
server.name: "<redacted>"

elasticsearch.url: "https://<redacted>:9200"

elasticsearch.username: "kibana"
elasticsearch.password: "<redacted>"

server.ssl.enabled: true
server.ssl.certificate: /etc/kibana/certs/wildcard.crt
server.ssl.key: /etc/kibana/certs/wildcard.key

elasticsearch.ssl.certificateAuthorities: /etc/kibana/certs/elastic-stack-ca.crt.pem
elasticsearch.ssl.verificationMode: certificate

pid.file: /var/run/kibana/kibana.pid

logging.dest: /data/kibana/logs/kibana.log

logging.verbose: true

xpack.security.enabled: true
xpack.monitoring.enabled: true
xpack.reporting.queue.timeout: 600000
xpack.reporting.csv.maxSizeBytes: 104857600
xpack.reporting.roles.allow: ["infrastructure"]
xpack.apm.ui.enabled: false

xpack.security.public:
  protocol: https
  hostname: <redacted>
  port: 443

xpack.security.sessionTimeout: 43200000
xpack.security.authProviders: [saml, basic]
server.xsrf.whitelist: [/api/security/v1/saml]
```

I restarted the kibana service after updating the `action.auto_create_index` setting on all of the nodes. Just to be sure, I stopped and started the kibana service, just now, and the `.managment-beats` index didn't get created when I tried to enroll a beat.

Let me know if you need anything else.

**Comment by mattapperson** [comment]
@thomas-mullaly Just checking that your kibana user has beats_admin role, or else superuser role assigned to it?

**Comment by thomas-mullaly** [comment]
@mattapperson Our `kibana` user doesn't have the `beats_admin` role, but it's also the built-in kibana account, so the UI doesn't let us change what roles it has (it just has `kibana_system`).

I switched our kibana instance to use a different account that's both `superuser` and `beats_admin` and the index still didn't get created when trying to enroll a beat. I did start and stop kibana after changing what user it connects to elasticsearch with.
